components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -193,72 +179,30 @@ class AutoPushFeature(
     }
 
     /**
-     * Register to receive push subscriptions when requested or when they have been re-registered.
-     *
-     * @param observer the observer that will be notified.
-     * @param owner the lifecycle owner for the observer. Defaults to [ProcessLifecycleOwner].
-     * @param autoPause whether to stop notifying the observer during onPause lifecycle events.
-     * Defaults to false so that subscriptions are always delivered to observers.
-     */
-    fun registerForSubscriptions(
-        observer: PushSubscriptionObserver,
-        owner: LifecycleOwner = ProcessLifecycleOwner.get(),
-        autoPause: Boolean = false
-    ) = subscriptionObservers.register(observer, owner, autoPause)
-
-    /**
-     * Register to receive push messages for the associated [PushType].
+     * Subscribes and notifies observers about the subscription information if available.
      *
-     * @param type the push message type that you want to be registered.
-     * @param observer the observer that will be notified.
-     * @param owner the lifecycle owner for the observer. Defaults to [ProcessLifecycleOwner].
-     * @param autoPause whether to stop notifying the observer during onPause lifecycle events.
-     * Defaults to false so that messages are always delivered to observers.
+     * @param scope The subscription identifier which usually represents the website's URI.
      */
-    fun registerForPushMessages(
-        type: PushType,
-        observer: Bus.Observer<PushType, String>,
-        owner: LifecycleOwner = ProcessLifecycleOwner.get(),
-        autoPause: Boolean = false
-    ) = messageObserverBus.register(type, observer, owner, autoPause)
-
-    /**
-     * Notifies observers about the subscription information for the push type if available.
-     */
-    fun subscribeForType(type: PushType) {
+    fun subscribe(scope: String, appServerKey: String? = null, onSubscribe: ((AutoPushSubscription) -> Unit) = {}) {
         DeliveryManager.with(connection) {
-            scope.launchAndTry {
-                val sub = subscribe(type.toChannelId()).toPushSubscription()
-                subscriptionObservers.notifyObservers { onSubscriptionAvailable(sub) }
+            coroutineScope.launchAndTry {
+                val sub = subscribe(scope, appServerKey)
+                onSubscribe(sub)
```

Comment:
same comment as for `unsubscribe` - don't we also want a `onError` callback here?

Reply:
Should be the same as the comment above.